# Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

In [1]:
from math import factorial

In [2]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [3]:
def arrangements(n, k):
    return int(factorial(n) / factorial(n - k))

In [4]:
def permutations(n):
    return int(factorial(n))

## 1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

### а) 

Найдём сколькими способами можно выбрать из колоды, состоящей из 52 карт, 4 карты?

Для ответа на этот вопрос воспользуемся формулой для вычисления количества сочетаний:

## $$C_{52}^4 = \frac{52!}{4!(52 - 4)!}$$

In [5]:
combinations(52, 4)

270725

Найдём сколькими способами можно выбрать из колоды, состоящей из 52 карт, 4 карты масти крести?

Количество карт масти крести равно $\frac{52}{4} = 13$

Для ответа на этот вопрос воспользуемся формулой для вычисления количества сочетаний:

## $$C_{13}^4 = \frac{13!}{4!(13 - 4)!}$$

In [6]:
combinations(13, 4)

715

Вероятность того, что все карты – крести:

In [7]:
715/270725

0.0026410564225690276

### б) 

Рассмотрим ситуацию, противоположную описанному событию, когда среди 4 выбранных карт нет ни одного туза.

Четыре карты выбираются из 48 карт (без тузов), число таких вариантов:

In [8]:
combinations(48, 4)

194580

Вероятность того, что среди 4-х карт не окажется ни одного туза:

In [9]:
194580/270725

0.7187367254594146

Потому как это событие противоположное, то вероятность исходного события равна:

In [10]:
1 - (194580/270725)

0.2812632745405854

Другой вариант решения задачи - последовательно рассмотреть ситуации, когда выбирается:
- Один туз из четырех карт;
- Два туза из четырех карт;
- Три туза из четырех карт;
- Четыре туза из четырех карт.

1) Сначала рассмотрим ситуацию, когда выбирается один туз из четырех. Число вариантов выбора равно 4.

Остальные три карты выбираются из 48 карт — тузы не рассматриваются, так как они уже выбраны:

In [11]:
combinations(48, 3)

17296

И в этом случае будет такое число сочетаний, когда из колоды, в которой 52 картs, выбраны четыре, одна из которых — туз:

In [12]:
4 * 17296

69184

2) Рассмотрим ситуацию, когда выбирается два туза из четырех. Число таких сочетаний будет равно:

In [13]:
combinations(4, 2)

6

Остальные две карты выбираются из 48 карт — тузы не рассматриваются, так как они уже выбраны:

In [14]:
combinations(48, 2)

1128

И в этом случае будет такое число сочетаний, когда из колоды, в которой 52 картs, выбраны четыре, две из которых — туз:

In [15]:
6 * 1128

6768

3) Рассмотрим ситуацию, когда выбирается три туза из четырех. Число таких сочетаний будет равно:

In [16]:
combinations(4, 3)

4

Оставшаяся карта выбираются из 48 карт — тузы не рассматриваются, так как они уже выбраны. Число вариантов выбора = 48

И в этом случае будет такое число сочетаний, когда из колоды, в которой 52 картs, выбраны четыре, три из которых — туз:

In [17]:
4 * 48

192

4) Рассмотрим ситуацию, когда выбирается четыре туза из четырех. Такой вариант выбора всего 1.

В итоге общее число благоприятных исходов равно:

In [18]:
69184 + 6768 + 192 + 1

76145

Вероятность того, что среди 4-х карт окажется хотя бы один туз:

In [19]:
76145/270725

0.2812632745405855

### __Ответ__: а) 0.0026410564225690276; б) 0.2812632745405855

## 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Найдём сколькими способами можно нажать 3 кнопки?

Для ответа на этот вопрос воспользуемся формулой для вычисления количества сочетаний:

## $$C_{10}^3 = \frac{10!}{3!(10 - 3)!}$$

In [20]:
combinations(10, 3)

120

Вероятность того, что человек, не знающий код, откроет дверь с первой попытки:

In [21]:
1/120

0.008333333333333333

### __Ответ__: 0.008333333333333333

## 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

#### Первый способ

Вероятность того, вероятность того, что первая извлеченная деталь окажется окрашена:

In [22]:
9/15

0.6

Вероятность того, что вторая извлеченная деталь окажется окрашена, при условии что первая извлеченная деталь была окрашена:

In [23]:
8/14

0.5714285714285714

Вероятность того, что третья извлеченная деталь окажется окрашена, при условии что первая и вторая извлеченные детали были окрашены:

In [24]:
7/13

0.5384615384615384

Вероятность одновременного появления всех трёх событий равна произведению их вероятностей:

In [25]:
(9/15)*(8/14)*(7/13)

0.18461538461538457

#### Второй способ

Найдём число сочетаний благоприятных исходов - извлечения 3 окрашенных деталей из 9:

In [26]:
combinations(9, 3)

84

 Число всех возможных исходов извлечения 3 деталей из 15:

In [27]:
combinations(15, 3)

455

 Искомая вероятность:

In [28]:
84/455

0.18461538461538463

### __Ответ__: 0.18461538461538463

## 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

#### Первый способ

Вероятность того, вероятность того, что первый приобретенный билет окажется выигрышным:

In [29]:
2/100

0.02

Вероятность того, вероятность того, что второй приобретенный билет окажется выигрышным, при условии что один выигрышный уже приобрели:

In [30]:
1/99

0.010101010101010102

Вероятность того, вероятность того, что 2 приобретенных билета окажутся выигрышными:

In [31]:
(2/100)*(1/99)

0.00020202020202020205

#### Второй способ

Найдём сколькими способами можно купить 2 билета из 100?

Для ответа на этот вопрос воспользуемся формулой для вычисления количества сочетаний:

## $$C_{100}^2 = \frac{100!}{2!(100 - 2)!}$$

In [32]:
combinations(100, 2)

4950

Вероятность того, вероятность того, что 2 приобретенных билета окажутся выигрышными:

In [33]:
1/4950

0.00020202020202020202

### __Ответ__: 0.00020202020202020202